python/GafferImageTest/ImageReaderTest.py

```diff
@@ -561,5 +573,317 @@ def testWithChannelTestImage( self ):
+			[ "v2/Stereo/composited.exr", [
+					[ "left", "RGBAZ", ((1,1), (1919,1079)), (0.0782152, 0.0939382, 0.0316098, 0.769054), (0, 0, 0, 0), (0.600586, 0.593262, 0.344971, 1) ],
+					[ "right", "RGBAZ", ((1,1), (1919,1079)), (0.0725963, 0.0852973, 0.0297889, 0.78676), (0, 0, 0, 0), (0.600586, 0.591309, 0.345215, 1) ] ] ],
+			[ "v2/Stereo/Ground.exr", [
+					[ "left", "RGBAZ", ((0,0), (1920,741)), (0.122791, 0.0876847, 0.0307255, 0.915993), (0, 0, 0, 0), (0.352295, 0.229492, 0.0673485, 1) ],
+					[ "right", "RGBAZ", ((0,0), (1920,741)), (0.119776, 0.0856502, 0.0300023, 0.913899), (0, 0, 0, 0), (0.352295, 0.229492, 0.0666981, 1) ] ] ],
+			[ "v2/Stereo/Leaves.exr", [
+					[ "left", "RGBAZ", ((0,0), (1920,1080)), (0.0325196, 0.0705669, 0.0221435, 0.429987), (0, 0, 0, 0), (0.40332, 0.593262, 0.344971, 1) ],
+					[ "right", "RGBAZ", ((0,0), (1920,1080)), (0.0297268, 0.064128, 0.0210256, 0.463715), (0, 0, 0, 0), (0.402832, 0.591309, 0.345215, 1) ] ] ],
+			[ "v2/Stereo/Trunks.exr", [
+					[ "left", "RGBAZ", ((0,0), (1920,814)), (0.0072465, 0.00569297, 0.00303837, 0.0964944), (0, 0, 0, 0), (0.397461, 0.307129, 0.177368, 1) ],
+					[ "right", "RGBAZ", ((0,0), (1883,814)), (0.00711977, 0.00569962, 0.00315493, 0.111899), (0, 0, 0, 0), (0.396729, 0.306396, 0.17749, 1) ] ] ]
+		]:
+			reader["fileName"].setValue( directory + name )
+
+			# The EXR examples should load correctly with either our default interpretation, or a strict
+			# interpretation of the spec
+			for interpretation in [ GafferImage.ImageReader.ChannelInterpretation.Default, GafferImage.ImageReader.ChannelInterpretation.Specification ]:
+				reader["channelInterpretation"].setValue( interpretation )
+				self.assertEqual( list( reader["out"].viewNames() ), [ i[0] for i in expected ] if expected[0][0] else []  )
+				for e in expected:
+					channelNames = reader["out"].channelNames( e[0] )
+					self.assertEqual( "".join( channelNames ), e[1] )
+
+					dw = reader["out"].dataWindow( e[0] )
+					self.assertEqual( dw, imath.Box2i( imath.V2i( *e[2][0] ), imath.V2i( *e[2][1] ) ) )
+
+					stats["view"].setValue( e[0] or "" )
+					if list( channelNames ) == ["Y"]:
+						stats['channels'].setValue( IECore.StringVectorData( ["Y", "Y", "Y", "A" ] ) )
+					else:
+						stats['channels'].setValue( IECore.StringVectorData( ["R", "G", "B", "A" ] ) )
+					
+					stats["area"].setValue( dw )
+					for i in range( 4 ):
+						# TODO - It's even more awful than usual for Python that the standard spec for
+						# assertAlmostEqual doesn't use significant digits.  Should really put a proper
+						# version of assertAlmostEqual in GafferTest.TestCase ( we could handle imath types
+						# as well ), and also handle nans properly
+						if not ( stats["average"].getValue()[i] != stats["average"].getValue()[i] and e[3][i] != e[3][i] or abs( e[3][i] ) > 1e20 ):
```

Comment:
I'm using an extremely imprecise tolerance of `places = 4`, because specifying a number of places is a terrible way of setting tolerance, if the values can range widely ( some of the test images contain large values ).  A more precise and meaningful test would be that values are within 6 significant digits

Reply:
It was the conditional guarding the `assertAlmostEqual()` that I wasn't following...

Reply:
The EXR examples try to exercise corner cases, so they include very large numbers ( which would never pass unless we raise the tolerance so much it made the other tests meaningless, because of tolerance being specified badly ), and NaNs ( assertAlmostEqual fails on Nans, and throws if you try to assert Nan == Nan ) - the conditional skips these cases where  assertAlmostEqual doesn't work